### ISSUE TITLE
Get rid of references and code specific to Python 2.5

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5715]
|[<img alt="vperic's avatar" src="https://avatars.githubusercontent.com/u/688281?s=50" width="50" height="50">](https://github.com/vperic)| @vperic reported|
|-|-|
|Trac ID|trac#5715|
|Type|task|
|Created|2012-06-19 17:31:31Z|

After ticket [#5553](https://github.com/twisted/twisted/issues/5553) officially drops Python 2.5 support, we should get rid of code referencing Python 2.5 specifically. This should probably only be done after [#5386](https://github.com/twisted/twisted/issues/5386) (the same for Python 2.3) and [#5387](https://github.com/twisted/twisted/issues/5387) (2.4).

Some of these comments can be removed, some might require the removal of code as well; grep gives the following mentions of Python 2.5:

```
twisted/conch/checkers.py:18:    # Python 2.5 got spwd to interface with shadow passwords
twisted/internet/selectreactor.py:117:            # subclass, but on Python 2.5 and earlier it is not.)
twisted/python/filepath.py:340:            # Under Python 2.5 on Windows, WindowsError has a winerror
twisted/python/filepath.py:355:            # ERROR_PATH_NOT_FOUND.  However, in Python 2.5,
twisted/python/filepath.py:682:    precision, call C{os.stat_float_times(True)}, or use Python 2.5.
twisted/test/test_failure.py:857:    Tests C{failure.Failure} support for generator features added in Python 2.5
wisted/trial/test/test_assertions.py:832:            # Python 2.5 and earlier, stacklevel of 0 resulted in a warning
twisted/web/_flatten.py:314:            # In Python 2.5, after an exception, a generator's gi_frame is
```

And everything around Python <= 2.6:

```
twisted/__init__.py:16:    if version < (2, 6):
twisted/logger/_global.py:209:            if sys.version_info < (2, 6):
twisted/logger/test/test_flatten.py:149:            if sys.version_info[:2] == (2, 6):
twisted/python/zippath.py:21:if sys.version_info[:2] >= (2, 6):
twisted/test/test_log.py:261:        if sys.version_info >= (2, 6):
twisted/test/test_rebuild.py:195:    if sys.version_info < (2, 6):
twisted/test/test_rebuild.py:213:    if sys.version_info >= (2, 6):
twisted/test/test_twisted.py:194:    supportedPythonVersion = (2, 6)
twisted/test/test_twisted.py:640:if sys.version_info[:2] == (2, 6):
twisted/topfiles/setup.py:12:if sys.version_info < (2, 6):
twisted/topfiles/setup.py:40:if sys.version_info[:2] <= (2, 6):
twisted/trial/_synctest.py:243:        if sys.version_info[:2] > (2, 6):
twisted/internet/kqreactor.py:17:This implementation depends on Python 2.6 or higher which has kqueue support
twisted/topfiles/setup.py:13:    print >>sys.stderr, "You must use at least Python 2.6 for Twisted"
```

<details><summary>Searchable metadata</summary>

```
trac-id__5715 5715
type__task task
reporter__vperic vperic
priority__lowest lowest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1340127091000000 1340127091000000
changetime__1503916221743206 1503916221743206
version__None None
owner__thijs thijs
cc__thijs cc__grimley
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

Update ticket description.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra commented|
|-|-|

Replying to [vperic](#5715):
> ./twisted/trial/test/test_assertions.py:724:            # Python 2.5 and earlier, stacklevel of 0 resulted in a warning

This seems like a test worth to keep around.

> ./twisted/web/_flatten.py:314:            # In Python 2.5, after an exception, a generator's gi_frame is
> ./twisted/python/util.py:873:    2.5's implementation of L{id} always returns positive values.

I've opened a new ticket for this so it can be deprecated at some later release (it's deprecated in 13.0 and 14+ seems a better time-frame to completely remove it).

> ./twisted/python/filepath.py:333:            # Under Python 2.5 on Windows, WindowsError has a winerror
> ./twisted/python/filepath.py:348:            # ERROR_PATH_NOT_FOUND.  However, in Python 2.5,

These two are reported in [#6627](https://github.com/twisted/twisted/issues/6627).

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra commented|
|-|-|

Replying to [thijs](#note_4):
> > ./twisted/python/util.py:873:    2.5's implementation of L{id} always returns positive values.
> 
> I've opened a new ticket for this so it can be deprecated at some later release (it's deprecated in 13.0 and 14+ seems a better time-frame to completely remove it).
> 

That's ticket [#6707](https://github.com/twisted/twisted/issues/6707).

**Comment by twisted-trac** [comment]
|<img alt="grimley's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|grimley set owner to grimley<br>grimley set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="grimley's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_7"></a>|grimley commented|
|-|-|

Should the NEWS file (twisted/web/topfiles/NEWS) be left as is.  Python support is added in Version 0.7.0 It seems unreasonable to remove this in as a place for historical tracking of changes.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_9"></a>|@thijstriemstra commented|
|-|-|

update grep results (twisted 15.2.1).

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_10"></a>|@thijstriemstra commented|
|-|-|

add python < 2.6 stuff

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_11"></a>|@rodrigc set status to `closed`|
|-|-|

Fixed in multiple tickets as Python 2.6 support was dropped,
and Python 3 support was added.
